### ISSUE TITLE
MacOS - SafetensorError: Error while deserializing header: HeaderTooLarge

### ISSUE THREAD

**Issue by tajchert** [microsoft/OmniParser#18]
Hi, 

I want to run on MacOS (Apple silicon), my steps so far:
``` bash
git clone git@github.com:microsoft/OmniParser.git
cd OmniParser/
pip install -r requirement.txt
cd weights/
git clone git@hf.co:microsoft/OmniParser
mv OmniParser/icon_caption_blip2 ../
mv OmniParser/icon_caption_florence ../
cd ..
pip install einops
python gradio_demo.py
```

Error I'm getting is:
``` bash
/Users/{user}/.pyenv/versions/3.12.6/lib/python3.12/site-packages/ultralytics/nn/tasks.py:714: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  ckpt = torch.load(file, map_location="cpu")
/Users/{user}/.pyenv/versions/3.12.6/lib/python3.12/site-packages/timm/models/layers/__init__.py:48: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
Traceback (most recent call last):
  File "/Users/{user}/coding/ai/OmniParser/gradio_demo.py", line 16, in <module>
    caption_model_processor = get_caption_model_processor(model_name="florence2", model_name_or_path="weights/icon_caption_florence")
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/{user}/coding/ai/OmniParser/utils.py", line 54, in get_caption_model_processor
    model = AutoModelForCausalLM.from_pretrained(model_name_or_path, torch_dtype=torch.float32, trust_remote_code=True)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/{user}/.pyenv/versions/3.12.6/lib/python3.12/site-packages/transformers/models/auto/auto_factory.py", line 559, in from_pretrained
    return model_class.from_pretrained(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/{user}/.pyenv/versions/3.12.6/lib/python3.12/site-packages/transformers/modeling_utils.py", line 3993, in from_pretrained
    with safe_open(resolved_archive_file, framework="pt") as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
safetensors_rust.SafetensorError: Error while deserializing header: HeaderTooLarge
```

I'm not sure what is the root cause of this issue as is safetensors should run correctly on Apple M-series chips (arm64).

**Comment by nelliedoll** [comment]
Stay off my shit

**Comment by tajchert** [comment]
This is solved for me.
I was trying to run it without `python weights/convert_safetensor_to_pt.py` as README file didn't have it (at the time of opening the issue). This was fixed in commit #8a3c07d
